eoreader/products/optical/vhr_product.py

```diff
@@ -515,8 +515,8 @@ def _get_path(self, filename: str = "", extension: str = "") -> AnyPathType:
             else:
                 prod_path = next(self.path.glob(f"{filename}*.{extension}"))
 
-        except (FileNotFoundError, IndexError):
-            LOGGER.warning(
+        except (FileNotFoundError, IndexError, StopIteration):
+            raise InvalidProductError(
```

Comment:
raise an exception instead of LOGGER. To confirm with @remi-braun

Reply:
lessgo 🚀